CRM/Contribute/Form/ContributionView.php

```diff
@@ -46,6 +46,7 @@ public function preProcess() {
     }
     $this->assign('is_template', $values['is_template']);
 
+    $noACL = FALSE;
     if (CRM_Financial_BAO_FinancialType::isACLFinancialTypeStatus() && $this->_action & CRM_Core_Action::VIEW) {
```

Comment:
This section should wind up removable with the view bounce being done [here](https://github.com/civicrm/civicrm-core/pull/22684/files) and then all the other checks can be done with the `checkAccess` being used further down

![image](https://user-images.githubusercontent.com/336308/152266648-652c37e9-18b1-4123-a2f4-9c07ef5efc22.png)